### ISSUE TITLE
Is the check for PR existence still needed now that we read the body of 422 response?

### ISSUE THREAD

**Issue by PawelLipski** [PawelLipski/git-machete#342]
`git_machete.github.create_pull_request`:

```
    prs: List[GitHubPullRequest] = derive_pull_requests(org, repo)
    pr_found: Optional[GitHubPullRequest] = find_or_none(lambda pr: pr.base == base and pr.head == head, prs)
    if not pr_found:
        pr = __fire_github_api_request('POST', f'/repos/{org}/{repo}/pulls', token, request_body)
        return __parse_pr_json(pr)
    else:
        raise MacheteException(f'Pull request for branch {head} is already created under link {pr_found.html_url}!\nPR details: {pr_found}')
```

Maybe we can skip one request to the API (`derive_pull_requests`) and rely on our new handling for 422 now?

**Comment by rifel123** [comment]
While i was testing extracting message body from `UnprocessableEntityError` i had this check disabled, so the output is pretty much same as above `Pull request for branch {head} is already created under link {pr_found.html_url}!\nPR details: {pr_found}'`, see:
<img width="874" alt="Screenshot 2021-11-08 at 09 08 55" src="https://user-images.githubusercontent.com/73847352/141771423-7a33ae2d-dd03-402a-91ca-d9acfc0d445c.png">
Please note that message regarding opening a ticket is already removed.

So I think, yes, this check is not needed anymore.

**Comment by rifel123** [comment]
closed with #350
